cli/azd/pkg/apphost/generate.go

```diff
@@ -900,17 +917,21 @@ func (b infraGenerator) evalBindingRef(v string, emitType inputEmitType) (string
 		default:
 			return "", errUnsupportedProperty("azure.servicebus.v0", prop)
 		}
-	case targetType == "azure.appinsights.v0":
+	case targetType == "azure.appinsights.v0" ||
+		targetType == "azure.sql.database.v0" ||
+		targetType == "sqlserver.database.v0" ||
+		targetType == "sqlserver.database.v1":
 		switch prop {
 		case "connectionString":
 			return fmt.Sprintf("{{ .Env.SERVICE_BINDING_%s_CONNECTION_STRING }}", scaffold.AlphaSnakeUpper(resource)), nil
```

Comment:
Can we switch/move using `{{ .Env.SERVICE_BINDING_%s_CONNECTION_STRING }}` to using `return fmt.Sprintf(`%s"%s" }}`, containerAppSecretConnectionString, resource), nil` (like azure.cosmosdb.account.v0).

Here's why:  Right now, the connection String does not have a key or password (just b/c we use EntraId-only), so we are thinking it is fine to just set it as part of the containerApp Environnement.  However, as a customer, if I want to change from using EntraID only, to a classic admin user/password, I can run `azd infra synth` and update bicep definition to use a classic admin user/password.  For that case, azd will keep the new connectionString (now with user and pass) secured.